wemake_python_styleguide/logic/tree/classes.py

```diff
@@ -70,35 +72,40 @@ def get_attributes(
         A tuple of lists for both class and instance level variables.
 
     """
-    class_attributes = []
-    instance_attributes = []
+    class_assignments = []
+    instance_assignments = []
 
     for subnode in ast.walk(node):
-        instance_attr = _get_instance_attribute(subnode)
-        if instance_attr is not None:
-            instance_attributes.append(instance_attr)
+        instance_assign = _get_instance_assignment(subnode)
+        if instance_assign is not None:
+            instance_assignments.append(instance_assign)
             continue
 
         if include_annotated:
-            class_attr = _get_annotated_class_attribute(node, subnode)
+            class_assign = _get_annotated_class_attribute(node, subnode)
         else:
-            class_attr = _get_class_attribute(node, subnode)
-        if class_attr is not None:
-            class_attributes.append(class_attr)
+            class_assign = _get_class_assignment(node, subnode)
+        if class_assign is not None:
+            class_assignments.append(class_assign)
 
-    return class_attributes, instance_attributes
+    return class_assignments, instance_assignments
 
 
-def _get_instance_attribute(node: ast.AST) -> Optional[ast.Attribute]:
-    return node if (
-        isinstance(node, ast.Attribute) and
-        isinstance(node.ctx, ast.Store) and
-        isinstance(node.value, ast.Name) and
-        node.value.id == 'self'
+def _get_instance_assignment(subnode: ast.AST) -> Optional[AnyAssign]:
+    return subnode if (
+        isinstance(subnode, AssignNodes) and
+        any(
+            isinstance(target, ast.Attribute) and
+            isinstance(target.ctx, ast.Store) and
+            isinstance(target.value, ast.Name) and
+            target.value.id in constants.SPECIAL_ARGUMENT_NAMES_WHITELIST
```

Comment:
This is not correct, because `constants.SPECIAL_ARGUMENT_NAMES_WHITELIST` also has `mcs` and `cls`, so it can catch non-self assigns.